webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -92,24 +111,56 @@ const ExperimentCheckpoints: React.FC<Props> = ({ experiment, pageRef }: Props)
     />
   ), [ handleStateFilterApply, handleStateFilterReset, settings.state ]);
 
-  const handleRegisterCheckpoint = useCallback((checkpoint: CoreApiGenericCheckpoint) => {
-    openModalCheckpointRegister({ checkpointUuid: checkpoint.uuid });
+  const handleRegisterCheckpoint = useCallback((checkpoints: string[]) => {
+    openModalCheckpointRegister({ checkpoints });
   }, [ openModalCheckpointRegister ]);
 
-  const dropDownOnTrigger = useCallback((checkpoint: CoreApiGenericCheckpoint) => {
-    return { [checkpointAction.Register]: () => handleRegisterCheckpoint(checkpoint) };
-  }, [ handleRegisterCheckpoint ]);
+  const handleDeleteCheckpoint = useCallback((checkpoints: string[]) => {
+    openModalCheckpointDelete({ checkpoints });
+  }, [ openModalCheckpointDelete ]);
+
+  const dropDownOnTrigger = useCallback((checkpoints: string | string[]) => {
+    const checkpointsArr = Array.isArray(checkpoints) ? checkpoints : [ checkpoints ];
```

Comment:
It looks like if API failed for one checkpoint, the others aren't processed, so we would probably want to at least filter based on state here (state == completed for registered, state in [completed, errored] for delete?) to have best chance of successful API calls.

Reply:
That works for me. Are you sure that those are the only valid states for registering and deleting checkpoints?